### PR TITLE
Vector Functions for MatZ

### PR DESCRIPTION
**Pull request by jnsiemer**
This PR implements...
- [X] is_vector, is_row_vector, is_column_vector + has_single_entry
- [ ] norm_sqrd_eucl, norm_infty (subsequent PR)
- [ ] dot_product (subsequent PR)

including tests for `MatZ`.

Furthermore, I added a function `collect_entries` that efficiently collects all matrix entries without cloning any objects (incl. test).

### CODE REVIEW

File: src/integer/mat_z/vector/is_vector.rs [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+    /// use math::integer::MatZ;
+    /// use std::str::FromStr;
+    ///
+    /// let vec = MatZ::from_str("[[1]]").unwrap();
+    ///
+    /// assert!(vec.has_single_entry());
+    /// ```
+    pub fn has_single_entry(&self) -> bool {
+        self.is_column_vector() && self.is_row_vector()
+    }
```
Comment (Marvin-Beckmann) [inline src/integer/mat_z/vector/is_vector.rs:84]: I do not quite get the pupose of this function, but if yout think that we need it, feel free to keep it

**Review by Marvin-Beckmann — APPROVED** [review]
Looks good. Please add an example for the vectors to the examples of the MatZ struct. This can also be included in a following PR

File: src/integer/mat_z/vector/is_vector.rs (hunk 1 above)
Comment (jnsiemer) [reply src/integer/mat_z/vector/is_vector.rs:84]: This function could become interesting, e.g. we want to automatically unpack single entry matrices for arithmetic operations. But yes, currently, we do not have any use cases.

**Comment by jnsiemer** [comment]
> Looks good. Please add an example for the vectors to the examples of the MatZ struct. This can also be included in a following PR

Good idea. I will do that in a following PR.

**Pull request merged**